OpenSim/Simulation/GeodesicWrap/ImplicitSphere.h

```diff
@@ -0,0 +1,33 @@
+#ifndef IMPLICIT_SPHERE_H_
+#define IMPLICIT_SPHERE_H_
+
+#include "ImplicitSurfaceParams.h"
+
+namespace OpenSim
+{
+
+//=============================================================================
+// CONCRETE IMPLICIT SPHERICAL SURFACE PARAMETERS
+//=============================================================================
+// The minimal requirements for a specific implicit wrap surface
+// characterization: i.e. Cylinder, Sphere, etc.
+class ImplicitSphere final: public ImplicitSurfaceParamsImpl
```

Comment:
The `Sphere` model component that the user interacts with would be helpful to sketch out. I believe it would need the following:
- The `Frame` that the geometry is connected to (via a `Socket`).
- Whether or not the parametric, implicit, and/or analytic form is available.
- Any `Sphere`-specific parameters (i.e., the radius).

The first two features could live in a abstract base class `Surface`, similar to my sketch, but minus all those extra getters and setters.

Reply:
I think it would make sense for the `Surface` class to generate a `ImplicitSurfaceParamsImpl` that can be passed to a `GeodesicWrapObject` when constructing a `GeodesicPath`.

Reply:
> I think it would make sense for the `Surface` class to generate a `ImplicitSurfaceParamsImpl` that can be passed to a `GeodesicWrapObject` when constructing a `GeodesicPath`.

Ok so I just got some good feedback on my class design layout from @adamkewley (which is also tripping you up @nickbianco ;) ) So I went a bit nuts with putting virtual classes into boxes.

But for the split between parametric and implicit etc I thought it would still be nice:
The way I thought it would work is that a Surface component is able to create a `GeodesicPathSegment` to add to the total `GeometryPath`.
And we would design different classes: `ImplicitGeodesic` and `ParametricGeodesic` which both have `GeodesicPathSegment` as base.

A concrete `CylinderSurface` `Component` would then construct  `GeodesicPathSegment*` (base class), which is either the parametric or implicit one depending on a flag on the `CylinderSurface` component.

It still makes sense to me, but what do you think (@adamkewley, @nickbianco )?

Reply:
`GeodesicPathSegment` needs to hold a collection of `Surface`s, so it wouldn't make sense for surfaces to generate a path segment. I think we still need some sort of intermediate "wrapping obstacle" class which a surface can generate and add to a path segment.